eel/src/storage_persister.rs

```diff
@@ -558,127 +577,152 @@ where
+        remote_monitors.push(read_channel_monitor("tests/resources/monitors/state_1/1c4eb9c1d721ae7616ee480a735d175818510326701695ad4163aa00c6a320dd_0"));
+        remote_monitors.push(read_channel_monitor("tests/resources/monitors/state_1/e22f85d8e341ba2f0e9d069e6e9585a062b817553f4738264e3019f174b95612_0"));
+
+        let startup_variant = StoragePersister::infer_startup_variant::<InMemorySigner>(
+            &mut local_monitors,
+            &mut remote_monitors,
+        )
+        .unwrap();
+
+        assert_eq!(startup_variant, StartupVariant::Normal);
     }
 
-    impl RemoteStorage for StorageMock {
-        fn check_health(&self) -> bool {
-            self.storage.check_health()
-        }
+    #[test]
+    fn recovery_start() {
+        let mut local_monitors = Vec::new();
 
-        fn list_objects(&self, bucket: String) -> LipaResult<Vec<String>> {
-            Ok(self.storage.list_objects(bucket))
-        }
+        let mut remote_monitors = Vec::new();
+        remote_monitors.push(read_channel_monitor("tests/resources/monitors/state_1/1c4eb9c1d721ae7616ee480a735d175818510326701695ad4163aa00c6a320dd_0"));
+        remote_monitors.push(read_channel_monitor("tests/resources/monitors/state_1/e22f85d8e341ba2f0e9d069e6e9585a062b817553f4738264e3019f174b95612_0"));
 
-        fn object_exists(&self, bucket: String, key: String) -> LipaResult<bool> {
-            Ok(self.storage.object_exists(bucket, key))
-        }
+        let startup_variant = StoragePersister::infer_startup_variant::<InMemorySigner>(
+            &mut local_monitors,
+            &mut remote_monitors,
+        )
+        .unwrap();
 
-        fn get_object(&self, bucket: String, key: String) -> LipaResult<Vec<u8>> {
-            Ok(self.storage.get_object(bucket, key))
-        }
+        assert_eq!(startup_variant, StartupVariant::Recovery);
+    }
 
-        fn put_object(&self, bucket: String, key: String, value: Vec<u8>) -> LipaResult<()> {
-            self.storage.put_object(bucket, key, value);
-            Ok(())
-        }
+    #[test]
+    fn local_knows_about_1_more_channel() {
+        let mut local_monitors = Vec::new();
+        local_monitors.push(read_channel_monitor("tests/resources/monitors/state_3/1c4eb9c1d721ae7616ee480a735d175818510326701695ad4163aa00c6a320dd_0"));
+        local_monitors.push(read_channel_monitor("tests/resources/monitors/state_3/aa7e2798a1cdca1e0700b2e5f07fe82002b6d084a9758553c8310b9f702ed61f_0"));
+        local_monitors.push(read_channel_monitor("tests/resources/monitors/state_3/e22f85d8e341ba2f0e9d069e6e9585a062b817553f4738264e3019f174b95612_0"));
+
+        let mut remote_monitors = Vec::new();
+        remote_monitors.push(read_channel_monitor("tests/resources/monitors/state_2/1c4eb9c1d721ae7616ee480a735d175818510326701695ad4163aa00c6a320dd_0"));
+        remote_monitors.push(read_channel_monitor("tests/resources/monitors/state_2/e22f85d8e341ba2f0e9d069e6e9585a062b817553f4738264e3019f174b95612_0"));
```

Comment:
Does that still work as intended now that remote storage encryption has been merged? Also in the other test cases?

Reply:
Nope, nice catch; I should've thought of that before. I'll fix it today.

Reply:
>Nope, nice catch; I should've thought of that before. I'll fix it today.

Forget that. It did work as intended 😅 I thought it would break because the committed remote states aren't encrypted, but I had forgotten that I had the function `read_channel_monitor()` written specifically for these tests, which doesn't try to decrypt anything..